security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -584,21 +584,32 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 		// Re-generate secret if it's expired.
 		if sc.shouldRotate(&secret) {
 			atomic.AddUint64(&sc.secretChangedCount, 1)
-
-			// if user not set PROV_CERT Directory Path for client to extract certs from path
-			// we will first check whether the token is expired or not
-			if sc.secOpts.ProvCert == "" && sc.isTokenExpired(&secret) {
+			// Send the notification to close the stream if token is expired, so that client could re-connect with a new token.
+			if isTokenExpired(secret.Token) && !sc.useCertToRotate() {
 				cacheLog.Debugf("%s token expired", logPrefix)
-				// TODO(myidpt): Optimization needed. When using local JWT, server should directly push the new secret instead of
-				// requiring the client to send another SDS request.
-				sc.callbackWithTimeout(connKey, nil /*nil indicates close the streaming connection to proxy*/)
-				return true
+				// TODO (liminw): CredFetcher is a general interface. In 1.7, we limit the use on GCE only because
+				// GCE is the only supported plugin at the moment.
+				if sc.secOpts.CredFetcher != nil && sc.secOpts.CredFetcher.GetType() == security.GCE {
```

Comment:
I don't think you need the second check. If CredFetcher is set - it should be called.

In general - I don't think GetType should ever be called and used with ifs in any of the code ( and probably shouldn't even exist ).

Reply:
Removed the check for GetType().